product_to_scale_bizerba_extended/models/product.py

```diff
@@ -118,14 +119,13 @@ def write(self, vals):
                         defered[product.id] = 'write'
                 # ticking and unticking the "can be sold" checkbox
                 # trigger the corresponding product_scale_log
-                if 'sale_ok' in vals.keys() or 'active' in vals.keys():
-                    if ((product.sale_ok and product.active)
-                        and (not vals.get('sale_ok', False)
-                             or not vals.get('active', False))):
-                        product._send_to_scale_bizerba('unlink')
-                    elif ((product.sale_ok and vals.get('active', False))
-                          or (product.active and vals.get('sale_ok', False))):
+                if vals.get('active', product.active):
```

Comment:
```python
if 'active' in vals or product.active:
    ...
```
This is a more pythonesque and human readable way of writing this condition.

Reply:
you expression is not the same. I'll propose to keep the if expression like it is and keep it self explained